internal/codespaces/api/api.go

```diff
@@ -188,36 +189,54 @@ type CodespaceEnvironmentConnection struct {
 
 // ListCodespaces returns a list of codespaces for the user.
 func (a *API) ListCodespaces(ctx context.Context) ([]*Codespace, error) {
-	req, err := http.NewRequest(
-		http.MethodGet, a.githubAPI+"/user/codespaces", nil,
-	)
-	if err != nil {
-		return nil, fmt.Errorf("error creating request: %w", err)
-	}
+	page := 1
+	per_page := 50
```

Comment:
This per page has an inherent limit on the server-side right? If I pass 100, the server will ignore it and use 50?

Reply:
right now, it will use the default if the per_page is out of range. I'm going to create another PR that uses the max if we request more along with a couple other updates.